src/lib/comms/sol-lwm2m.c

```diff
@@ -4184,6 +4363,384 @@ sol_lwm2m_bootstrap_server_send_finish(struct sol_lwm2m_bootstrap_server *server
+    struct obj_ctx *obj_ctx,
+    uint16_t content_format, const struct sol_str_slice payload)
+{
+    struct sol_vector object_tlvs;
+    struct sol_lwm2m_tlv *instance_tlvs;
+    struct obj_instance *obj_instance;
+    uint16_t i;
+    int r;
+
+    if (!obj_ctx->obj->write_tlv || !obj_ctx->obj->create) {
+        SOL_WRN("Object %" PRIu16 " does not support the Bootstrap Write"
+            " method for multiple Object Instances",
+            obj_ctx->obj->id);
+        return SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
+    }
+
+    if (!content_format) {
+        SOL_WRN("Content format was not set."
+            " Impossible to write object");
+        return SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
+    }
+
+    if (!payload.len) {
+        SOL_WRN("Payload to write on object /%" PRIu16
+            " is empty", obj_ctx->obj->id);
+        return SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
+    }
+
+    if (content_format == SOL_LWM2M_CONTENT_TYPE_TLV) {
+        r = sol_lwm2m_parse_tlv(payload, &object_tlvs);
+        SOL_INT_CHECK(r, < 0, SOL_COAP_RESPONSE_CODE_BAD_REQUEST);
+
+        SOL_VECTOR_FOREACH_IDX (&object_tlvs, instance_tlvs, i) {
+            if (instance_tlvs->type != SOL_LWM2M_TLV_TYPE_OBJECT_INSTANCE) {
+                SOL_WRN("Only TLV is supported for writing an individual object instance."
+                    " Received: %" PRIu16 ". Skipping this instance.", instance_tlvs->type);
+                continue;
+            }
+
+            obj_instance = find_object_instance_by_instance_id(obj_ctx, obj_instance->id);
+
+            //Instance already exists?
+            if (obj_instance) {
+                // TODO: Call write_tlv callback
+                // TODO: How to build the tlvs parameter?
+            } else {
+                obj_instance = sol_vector_append(&obj_ctx->instances);
+                SOL_NULL_CHECK(obj_instance, SOL_COAP_RESPONSE_CODE_BAD_REQUEST);
+
+                obj_instance->id = instance_tlvs->id;
+
+                sol_vector_init(&obj_instance->resources_ctx, sizeof(struct resource_ctx));
+
+                // TODO: Call create callback
+                // TODO: How to build the payload parameter back from the tlv?
+
+                SOL_INT_CHECK_GOTO(r, < 0, err_exit);
+            }
+        }
+
+        sol_lwm2m_tlv_list_clear(&object_tlvs);
+        SOL_INT_CHECK(r, < 0, SOL_COAP_RESPONSE_CODE_BAD_REQUEST);
+
+        SOL_DBG("Bootstrap Write on object /%" PRIu16
+            " succeeded!", obj_ctx->obj->id);
+
+        return SOL_COAP_RESPONSE_CODE_CHANGED;
+    } else {
+        SOL_WRN("Only TLV is supported for writing multiple object instances."
+            " Received: %" PRIu16 "", content_format);
+        return SOL_COAP_RESPONSE_CODE_UNSUPPORTED_CONTENT_FORMAT;
+    }
+
+err_exit:
+    obj_instance_clear(client, obj_ctx, obj_instance);
+    return SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
+}
+
+static int
+handle_unknown_bootstrap_resource(void *data, struct sol_coap_server *server,
```

Comment:
Why did you create this function? Why not use `handle_resource` ?
This is pratically copy & pasta. You must adapt the `handle_resource` function in order to support the bootstrap feature.